fabtests/functional/stream_msg.c

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.  All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <getopt.h>
+#include "shared.h"
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <string.h>
+#include <rdma/fi_cm.h>
+
+const char *msg = "hello stream!";
+
+
+int send_stream(struct fid_ep *ep, const char *msg, size_t msg_len)
+{
+	int offset, ret;
+
+	for (offset = 0; offset < msg_len; ) {
+		ret = fi_send(ep, (msg + offset), (msg_len - offset), NULL, 0, NULL);
+		if (ret < 0 && ret != -FI_EAGAIN) {
+			fprintf(stderr, "%s error %s\n", __func__, fi_strerror(-ret));
+			return ret;
+		}
+
+		if (ret > 0)
```

Comment:
I think it is not clear when this validation becomes true, since ret comes from fi_send(), and it's either negative or zero.

Reply:
The rstream EP returns the number of bytes sent.